### PR TITLE
416 fix Webpack 5 issue with NodeJS libs + other goodies

### PR DESCRIPTION
**Pull request by leite08**
Ref: https://github.com/metriport/metriport-internal/issues/416

### Dependencies

none

### Description

- Currently, we've been deploying `alpha` versions in a way that sets the `latest` tag on the alpha version. This is not ideal since we're pointing users/cxs to dev versions of the libs.
    
   To solve this, we should use the `--dist-tag` option when calling Lerna. We can use the npm script `npm run publish:alpha` to do that. The form below does it, as well as automatically sets the to-be-publishes libs' version to increase on prerelease (alpha), accepts the default values by default, and provides the OTP (one time password):

   ```shell
   $ npm run publish:alpha -- prerelease --yes --otp xxxx
   ```

- Fix compilation with Webpack 5 not having polyfills for NodeJS libs, by using Craco to override Webpack's fallbacks - issue:

   Screenshot 2023-07-12 at 9 43 10 PM

- Export functions on CW SDK so we can import them w/o the folder/path.

- Add `declarationMap` to `tsconfig` so we can navigate to lib's code from the client.

- Fix `dockerfile.dev` + docker compose working on the updated monorepo - [context](https://metriport.slack.com/archives/C04DMKE9DME/p1689267697072979?thread_ts=1689262702.141299&cid=C04DMKE9DME)

### Release Plan

- nothing special

### CODE REVIEW

**Review by Orta21 — APPROVED** [review]

File: packages/api-sdk/src/shared.ts [hunk 1]
```diff
@@ -20,6 +20,6 @@ export const ISO_DATE = "YYYY-MM-DD";
 export const defaultString = z.string().trim();
 export const defaultOptionalString = optionalString(defaultString);
 export const defaultDateString = defaultString.refine(v => dayjs(v, ISO_DATE, true).isValid(), {
-  message: `Date must be a valid ISO 8601 date formatted ${ISO_DATE}. Example: 2023-05-03`,
+  message: `Date must be a valid ISO 8601 date formatted ${ISO_DATE}. Example: 2023-15-03`,
```
Comment (Goncharo) [inline packages/api-sdk/src/shared.ts:23]: shouldn't this be the other way around since it's YYYY-MM-DD?

ie `2023-03-15` or something

**Review by Goncharo — CHANGES_REQUESTED** [review]
generally LGTM - one thing that needs a fix

File: packages/api-sdk/src/shared.ts (hunk 1 above)
Comment (leite08) [reply packages/api-sdk/src/shared.ts:23]: good catch! 🕵️

**Review by Goncharo — APPROVED** [review]
:shipit: 

just a heads up about the PR check

**Pull request merged**